astore/server/main.go

```diff
@@ -210,34 +217,36 @@ func Start(targetURL, cookieDomain, oAuthType string, astoreFlags *astore.Flags,
 			http.Error(w, "invalid authorization path, tough luck, try again", http.StatusUnauthorized)
 			return
 		}
-		if err := authWeb.PerformLogin(w, r, oauth.WithState(*key), oauth.WithCookieOptions(kcookie.WithPath("/"))); err != nil {
+		if err := authWeb.PerformLogin(w, r,
+			oauth.WithState(*key),
+			oauth.WithCookieOptions(kcookie.WithPath("/")),
+		); err != nil {
 			http.Error(w, "oauth failed, no idea why, ask someone to look at the logs", http.StatusUnauthorized)
 			log.Printf("ERROR - could not perform login - %s", err)
 			return
 		}
 	})
 
-	// Path /e/ is the landing page at the end of the oauth authentication.
+	// Path /e/ is the landing page at the end of the oauth authentication. If the oauth landing page is a step in a multi-oauth flow,
+	// it will redirect to /a with additional logins.
 	mux.HandleFunc("/e/", func(w http.ResponseWriter, r *http.Request) {
 		copts := []kcookie.Modifier{kcookie.WithPath("/")}
 		if cookieDomain != "" {
 			// WithSecure and WithSameSite are required to get the cookie forwarded via the NASSH plugin in chrome (for SSH).
 			copts = append(copts, kcookie.WithDomain(cookieDomain), kcookie.WithSecure(true), kcookie.WithSameSite(http.SameSiteNoneMode))
 		}
-
-		data, handled, err := authWeb.PerformAuth(w, r, copts...)
+		data, _, err := authWeb.PerformAuth(w, r, copts...)
```

Comment:
why did you remove handled? The goal here is that if PerformAuth does a redirect, we don't call ShowResult below that can overwrite the http.Status and response returned.

(without checking handled here, ShowResult is always invoked, even after a redirect)